mapviz_plugins/include/mapviz_plugins/pointcloud2_plugin.h

```diff
@@ -0,0 +1,155 @@
+// *****************************************************************************
+//
+// Copyright (c) 2015, Southwest Research Institute® (SwRI®)
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Southwest Research Institute® (SwRI®) nor the
+//       names of its contributors may be used to endorse or promote products
+//       derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+// ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// *****************************************************************************
+
+#ifndef MAPVIZ_PLUGINS_POINTCLOUD2_PLUGIN_H_
+#define MAPVIZ_PLUGINS_POINTCLOUD2_PLUGIN_H_
+
+// C++ standard libraries
+#include <string>
+#include <deque>
+#include <vector>
+#include <map>
+
+#include <mapviz/mapviz_plugin.h>
+
+// QT libraries
+#include <QGLWidget>
+#include <QColor>
+
+// ROS libraries
+#include <sensor_msgs/PointCloud2.h>
+
+// QT autogenerated files
+#include "ui_pointcloud2_config.h"
+
+namespace mapviz_plugins
+{
+  class PointCloud2Plugin : public mapviz::MapvizPlugin
+  {
+    Q_OBJECT
+
+    public:
+      enum
+      {
+        COLOR_FLAT = 0,
+        COLOR_X = 1,
+        COLOR_Y = 2,
+        COLOR_Z = 3,
+        EXTRA_1 = 4,
+        EXTRA_2 = 5,
+        EXTRA_3 = 6,
+        EXTRA_4 = 7
+      };
```

Comment:
It looks like here and in a number of other places, the plugin is limited to handling a maximum of 4 different feature types (other than x/y/z); would it be possible to handle an arbitrary number of features?